### DISCUSSION TITLE
lb and lb4 are the same code?

### DISCUSSION THREAD

**Discussion by bitxplora** [loopbackio/loopback-next#10363] [Q&A] [answered]
I observed that `lb` and `lb4` are the same code, that is, the same code in two files with different names. What is the basis for this?

**Comment by achrinza** [comment]
Hi @bitxplora, could you clarify which package / where the `lb` command is from?

LoopBack 4 is current "version" of the project and uses the `lb4` command, which is provided by the `@loopback/cli` package.

**Reply by bitxplora** [reply]
When you installed loopback with `npm i -g @loopback/cli`, it will install both `lb` and `lb4` on your system, same codes and same work. To verify, do any of these:

Check the `bin` directory of your installed `node`.

OR

Do `which lb`, then `which lb4`.

Diff the two files.

**Reply by achrinza** [reply, marked as answer]
Thanks for the clarification - Indeed `@loopback/cli` does provide both `lb` and `lb4`.

However, both point to the same file:

https://github.com/loopbackio/loopback-next/blob/3e7b446e6273b368f1b29af3eb89ef530d1da6d0/packages/cli/package.json#L11-L14

We can verify this by running:

```sh
# blank output = no difference 
$ diff "$(which lb)" "$(which lb4)" 
```

Due to the way NPM installs CLI utilities, it will copy the same file twice to both `lb` and `lb4`.

This was added by @frbuceta in https://github.com/loopbackio/loopback-next/commit/f7ad0020fbadbf7125340cd2fd66336a1eae1b86. However I was not able to find a rationale for this change.

**Reply by bitxplora** [reply]
Thank you for your work on the project and your service to the community. 

I think `lb` is better, if there is no need to increase the appended number on every major release; `lb5`, `lb6`, etc.
